drivers/sensor/ltr390/ltr390_trigger.c

```diff
@@ -0,0 +1,351 @@
+/*
+ * Copyright (c) 2022 Luke Holt
+ *
+ * SPDX-Livense-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/byteorder.h>
+
+#include "ltr390.h"
+
+LOG_MODULE_DECLARE(LTR390, CONFIG_SENSOR_LOG_LEVEL);
+
+
+/**
+ * @brief Converts the raw bytes acquired from the sensor to an ambient light
+ *        value in lux.
+ *
+ * @param cfg Device config structure
+ * @param val Container to write the converted value to
+ * @param als_raw Buffer containing the raw bytes in LE
+ * @return int errno
+ */
+static inline int ltr390_als_value_to_bytes(
+	const struct ltr390_config *cfg,
+	const struct sensor_value *val,
+	void *als_raw)
+{
+	uint8_t *bytes = (uint8_t *)als_raw;
+	double l;
```

Comment:
This should be done with integer math rather than float math, take a look around the tree at other sensor drivers to get a sense of how you might go about that

Reply:
Done.